content/en/blog/_posts/2021-03-03-deletes-finaliers-and-owner-references.md

```diff
@@ -0,0 +1,270 @@
+}' \
+  -H "Content-Type: application/json"
+{
+  "kind": "Status",
+  "apiVersion": "v1",
+  "metadata": {},
+  "status": "Success",
+  "details": { ... }
+```
+
+Note that the propagation policy cannot be specified on the command line using kubectl. You have to specify it using a custom API call. Simply create a proxy, so you have access to the API server from the client, and execute a `curl` command with just a URL to execute that `delete` command. 
+
+There are three different options for the propagation policy:
+
+- `Foreground`: Children are deleted before the parent (post-order)
+- `Background`: Parent is deleted before the children (pre-order)
+- `Orphan`: Owner references are ignored
+
+Keep in mind that when you delete an object and owner references have been specified, finalizers will be honored in the process. This can result in trees of objects persisting, and you end up with a partial deletion. At that point, you have to look at any existing owner references on your objects, as well as any finalizers, to understand what’s happening. 
+
+## Forcing a Deletion of a Namespace
+
+There's one situation that may require forcing finalization for a namespace. If you've deleted a namespace and you've cleaned out all of the objects under it, but the namespace still exists, use the API call `finalize`.This informs the namespace controller that it needs to remove the finalizer from the namespace and perform any cleanup:
+
+```
+$ cat <<EOF | curl -X PUT \
+  localhost:8080/api/v1/namespaces/test/finalize \
+  -H "Content-Type: application/json" \
+  --data-binary @-
+{
+  "kind": "Namespace",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "test"
+  },
+  "spec": {
+    "finalizers": null,
+  }
+}
+EOF
+```
+
+This should be done with caution as it may delete the namespace only and leave orphan objects within the, now non-exiting, namespace - a confusing state for Kubernetes. If this happens, the namespace can be re-created manually and sometimes the orphaned objects will re-appear under the just-created namespace which will allow manual cleanup and recovery.
+
+## Key Takeaways
+
+As these examples demonstrate, finalizers can get in the way of deleting resources in Kubernetes, especially when there are parent-child relationships between objects. Often, there is a reason for adding a finalizer into the code, so you should always investigate before manually deleting it. Owner references allow you to specify and remove trees of resources, although finalizers will be honored in the process. Finally, the propagation policy can be used to specify the order of deletion via a custom API call, giving you control over how objects are deleted. Now that you know a little more about how deletions work in Kubernetes, we recommend you try it out on your own, using a test cluster. 
+
+All of the scenarios covered in this post are demonstrated in a recent webinar; you can [watch the recording here](https://www.youtube.com/watch?v=F7-ZxWwf4sY).

```

Comment:
If you want you can embed the video in the page, too. See https://gohugo.io/content-management/shortcodes/#youtube